landingpage/src/components/TransactionTool.tsx

```diff
@@ -0,0 +1,175 @@
+import { /*Transaction,*/ Psbt, script } from 'bitcoinjs-lib'
+import { bitcoinExplorer, PlebNameData, PlebNameHistory, util } from 'plebnames'
+import MarkedTextWithCopy from './MarkedTextWithCopy'
+import { useEffect, useState } from 'react'
+import DropDownContent from './HeaderWithSearch/DropDownContent'
+import AlterConfigForName, { AlterOptionKey } from './AlterConfigForName'
+
+interface TransactionToolProps {
+	mode: 'claimAndInscribe'|'inscribe'
+	name: string;
+	history?: PlebNameHistory;
+}
+
+type InscriptionMap = Partial<Record<AlterOptionKey, string>>;
+
+export const TransactionTool: React.FC<TransactionToolProps> = ({ name, mode, history }) => {
+	const [senderAddress, setSenderAddress] = useState(history?.getData().owner ??'')
+	const [validSenderAddress, setValidSenderAddress] = useState<string|undefined>(undefined)
+	const [senderUtxo, setSenderUtxo] = useState<bitcoinExplorer.UTXO|undefined>(undefined)
+	const [senderUtxoStatus, setSenderUtxoStatus] = useState<'addressNeeded'|'fetching'|'ok'|Error>('addressNeeded')
+	const [inscriptionMap, setInscriptionMap] = useState<InscriptionMap>({})
+	const [transaction, setTransaction] = useState<{
+		transaction: {toHex: () => string}
+		senderAddressError?: Error
+		senderUtxoError?: Error
+	} | undefined>(undefined)
+
+	useEffect(() => {
+		const tx = generateTransaction({name, senderAddress, senderUtxo, inscriptions: Object.values(inscriptionMap), mode})
```

Comment:
`Object.values(inscriptionMap)` only lists the values without the keys and "=".
Let's don't make things too complicated with `Partial<Record<AlterOptionKey, string>>` and just use `inscriptionsInAscii: string[]`

Reply:
recognized it is fine, as key is also encoded in value. But is misleading (at least for me). Does the TransactionTool really need to be aware of the key?

Reply:
Yes, it must be ensured that:
- there are no mutliple entries for the same key
- that only modified entries are added to `generateTransaction`  (not there yet)

Reply:
with only modified entries are added you mean preventing setting the same value again? e.g. setting "test.website=bitcoin.org" when it is already "test.website=bitcoin.org"?

I think these are mistakes that happen only rarely and the consequence is only to have spent more in fees than you need. Not sure if worth the implementation effort and complexity.

Reply:
that's a good question, either the output with the lowest index or the one with the highest index, I prefer a little bit the output with the highest index but maybe just the thing that is simpler to implement, what do you think? As long as the field is not 'owner' it is not critical.

Reply:
the current implementation is that the higher index overwrites the lower indexes, so the higher index goes into effect. Only for 'owner' the higher indexes are ignored as soon as the owner changes.